spec/system/admin/orders_spec.rb

```diff
@@ -571,28 +571,45 @@
             expect(page).to have_content "Confirmation emails sent for 2 orders."
           end
 
-          it "can bulk print invoices from 2 orders" do
-            page.find("#listing_orders tbody tr:nth-child(1) input[name='bulk_ids[]']").click
-            page.find("#listing_orders tbody tr:nth-child(2) input[name='bulk_ids[]']").click
+          shared_examples "can bulk print invoices from 2 orders" do |legal_invoices_enabled|
+            context "with legal invoices feature #{legal_invoices_enabled}" do
+              before do
+                if legal_invoices_enabled == "enabled"
+                  Flipper.enable(:invoices)
+                else
+                  Flipper.disable(:invoices)
+                end
+              end
 
-            page.find("span.icon-reorder", text: "ACTIONS").click
-            within ".ofn-drop-down .menu" do
-              expect {
-                page.find("span", text: "Print Invoices").click # Prints invoices in bulk
-              }.to enqueue_job(BulkInvoiceJob).exactly(:once)
-            end
+              it "can bulk print invoices from 2 orders" do
+                page.find("#listing_orders tbody tr:nth-child(1) input[name='bulk_ids[]']").click
+                page.find("#listing_orders tbody tr:nth-child(2) input[name='bulk_ids[]']").click
+
+                page.find("span.icon-reorder", text: "ACTIONS").click
+                within ".ofn-drop-down .menu" do
+                  expect {
+                    page.find("span", text: "Print Invoices").click # Prints invoices in bulk
+                  }.to enqueue_job(BulkInvoiceJob).exactly(:once)
+                end
+
+                expect(page).to have_content "Compiling Invoices"
+                expect(page).to have_content "Please wait until the PDF is ready " \
+                                             "before closing this modal."
 
-            expect(page).to have_content "Compiling Invoices"
-            expect(page).to have_content "Please wait until the PDF is ready " \
-                                         "before closing this modal."
+                # we don't run Sidekiq in test environment, so we need to manually run enqueued jobs
+                # to generate PDF files, and change the modal accordingly
+                perform_enqueued_jobs(only: BulkInvoiceJob)
 
-            # we don't run Sidekiq in test environment, so we need to manually run enqueued jobs
-            # to generate PDF files, and change the modal accordingly
-            perform_enqueued_jobs(only: BulkInvoiceJob)
+                expect(page).to have_content "Bulk Invoice created"
+                expect(page).to have_link(class: "button", text: "VIEW FILE", href: /invoices/)
```

Comment:
Hi @filipefurtad0 

Do we want to check the content of the pdf after this step? or do we want to do that in a separated file (as we did with `spec/system/admin/invoice_print_spec.rb`)?

Reply:
That's a good idea, yes.

I'm thinking what's the best approach to avoid repeating the whole test setup.:
- move the bulk printing test
- partially extract the test into a separate helper, an call it within `spec/system/admin/orders_spec.rb`

I'll experiment a bit around these options.